### PR TITLE
Add Pico14 board from Bolt Industries

### PR DESCRIPTION
**Pull request by davigupta**

### CODE REVIEW

**Comment by davigupta** [comment]
kmk_firmware$ python3 -m unittest
...............................................................
----------------------------------------------------------------------
Ran 63 tests in 15.140s

OK

File: boards/Pico14/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Pico 14 Numerical Keypad
+
+![Pico14](https://www.boltind.com/wp-content/uploads/2022/01/PXL_20220119_171113903-scaled.jpg)
+
+Pico 14 Numerical Keypad / Macro Pad PCB’s and Hardware Kit.
+
+kb.py is designed to work with the Pi Pico.
```
Comment (xs5871) [inline boards/Pico14/README.md:7]:
```suggestion
`kb.py` is designed to work with the Pi Pico.  
```

File: boards/Pico14/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Pico 14 Numerical Keypad
+
+![Pico14](https://www.boltind.com/wp-content/uploads/2022/01/PXL_20220119_171113903-scaled.jpg)
+
+Pico 14 Numerical Keypad / Macro Pad PCB’s and Hardware Kit.
+
+kb.py is designed to work with the Pi Pico.  
+
+Retailers
```
Comment (xs5871) [inline boards/Pico14/README.md:9]:
```suggestion
Retailers:
```

File: boards/Pico14/kb.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        board.GP19,
+        board.GP20,
+        board.GP21,
+        board.GP22,
+    )
+
+    diode_orientation = DiodeOrientation.COLUMNS
+
+    leds = LED(led_pin=[board.GP27], val=0)
+    _KMKKeyboard.extensions.append(leds)
```
Comment (xs5871) [inline boards/Pico14/kb.py:26]: This belongs into `main.py`.
Don't do code execution in the class header.

File: boards/Pico14/main.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    'NUMPAD_4', 'NUMPAD_5',     'NUMPAD_6',
+    'NUMPAD_1', 'NUMPAD_2',     'NUMPAD_3',
+    'NUMPAD_0', ______,         'NUMPAD_DOT',
+], [
+  # Layer 1
+    'NUMLOCK',  'NUMPAD_SLASH', 'NUMPAD_ASTERISK',
+    'HOME',     'UP',           'PGUP',
+    'LEFT',     'NUMPAD_5',     'RIGHT',
+    'END',      'DOWN',         'PGDN',
+    'INS',       ______,        'DEL',
```
Comment (xs5871) [inline boards/Pico14/main.py:46]: Please use the `KC.xxx` pattern for the default configuration. Stringy keymaps are a niche application for special cases and not the recommended for default use.

File: boards/Pico14/kb.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    row_pins = (
+        board.GP18,
+        board.GP19,
+        board.GP20,
+        board.GP21,
+        board.GP22,
+    )
+
+    diode_orientation = DiodeOrientation.COLUMNS
+    leds = LED(led_pin=[board.GP27], val=0)
```
Comment (xs5871) [inline boards/Pico14/kb.py:23]:
```suggestion
```

File: boards/Pico14/main.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+from kb import KMKKeyboard
+from kmk.extensions.lock_status import LockStatus
+from kmk.keys import KC
+from kmk.modules.layers import Layers
+
+Pico14 = KMKKeyboard()
+
```
Comment (xs5871) [inline boards/Pico14/main.py:7]:
```suggestion
leds = LED(led_pin=[board.GP27], val=0)

```

File: boards/Pico14/main.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+class LEDLockStatus(LockStatus):
+    def set_lock_leds(self):
+        if self.get_num_lock():
+            Pico14.leds.set_brightness(100, leds=[0])
```
Comment (xs5871) [inline boards/Pico14/main.py:11]:
```suggestion
            leds.set_brightness(100, leds=[0])
```

File: boards/Pico14/main.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        else:
+            Pico14.leds.set_brightness(0, leds=[0])
```
Comment (xs5871) [inline boards/Pico14/main.py:13]:
```suggestion
            leds.set_brightness(0, leds=[0])
```

File: boards/Pico14/main.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    def after_hid_send(self, sandbox):
+        super().after_hid_send(sandbox)  # Critically important. Removing this will break lock status.
+
+        if self.report_updated:
+            self.set_lock_leds()
+
+Pico14.modules.append(Layers())
+Pico14.extensions.append(Pico14.leds)
```
Comment (xs5871) [inline boards/Pico14/main.py:22]:
```suggestion
Pico14.extensions.append(leds)
```

File: boards/Pico14/kb.py (hunk 5 above)
Comment (xs5871) [reply boards/Pico14/kb.py:23]: This is still code execution in a class header.

**Review by xs5871 — CHANGES_REQUESTED** [review]

**Comment by xs5871** [comment]
Forgot to mention: once done, please consider squashing the history.

File: boards/Pico14/main.py (hunk 7 above)
Comment (davigupta) [reply boards/Pico14/main.py]: update a new diff

File: boards/Pico14/main.py (hunk 8 above)
Comment (davigupta) [reply boards/Pico14/main.py]: updated a new diff

File: boards/Pico14/main.py (hunk 9 above)
Comment (davigupta) [reply boards/Pico14/main.py]: update a new diff

**Review by davigupta — COMMENTED** [review]
Addressed review comments.

**Comment by xs5871** [comment]
Yep, that works. Now kindly clean up the history by rebasing/squashing, so we don't have a bunch of merge/fix commits in our history.

**Comment by davigupta** [comment]
Yes I squashed it. Pls see if you don’t see this in pull request.

Thanks

**Comment by xs5871** [comment]
Nope, still shows up as 16 individual commits.

**Comment by davigupta** [comment]
Ok I did squash only. Will do rebase and then squash.

Thanks

**Comment by davigupta** [comment]
Pls look at new pull request.

Thanks,
Davi

**Comment by davigupta** [comment]
Pls look a new pull request:

https://github.com/KMKfw/kmk_firmware/pull/674

**Comment by davigupta** [comment]
Thanks for the review. Can you pls take a look again?

https://github.com/KMKfw/kmk_firmware/pull/674

**Comment by davigupta** [comment]
Use old pull request 671 as requested.

**Review by xs5871 — APPROVED** [review]

**Comment by davigupta** [comment]
Thanks!

**Pull request merged**